packages/rs-drive-abci/src/execution/platform_events/identity_credit_withdrawal/update_broadcasted_withdrawal_transaction_statuses/v0/mod.rs

```diff
@@ -81,55 +104,67 @@ where
-                        Error::Execution(ExecutionError::CorruptedCodeExecution(
-                            "Can't get transactionId from withdrawal document",
-                        ))
-                    })?;
+                    .get_optional_identifier(withdrawal::properties::TRANSACTION_ID)?
+                    .ok_or(Error::Execution(ExecutionError::CorruptedDriveResponse(
+                        "Can't get transactionId from withdrawal document".to_string(),
+                    )))?;
 
                 let transaction_index = document
                     .properties()
-                    .get_integer(withdrawal::properties::TRANSACTION_INDEX)
+                    .get_optional_integer(withdrawal::properties::TRANSACTION_INDEX)?
+                    .ok_or(Error::Execution(ExecutionError::CorruptedDriveResponse(
+                        "Can't get transaction index from withdrawal document".to_string(),
+                    )))?;
+
+                let current_status: WithdrawalStatus = document
+                    .properties()
+                    .get_optional_integer::<u8>(withdrawal::properties::STATUS)?
+                    .ok_or(Error::Execution(ExecutionError::CorruptedDriveResponse(
+                        "Can't get transaction index from withdrawal document".to_string(),
+                    )))?
+                    .try_into()
                     .map_err(|_| {
-                        Error::Execution(ExecutionError::CorruptedCodeExecution(
-                            "Can't get transactionIdex from withdrawal document",
-                        ))
+                        Error::Execution(ExecutionError::CorruptedDriveResponse(format!(
+                            "Withdrawal status unknown"
+                        )))
                     })?;
 
-                let transaction_id = hex::encode(transaction_id_bytes);
-
                 let block_height_difference = block_execution_context
                     .block_state_info()
                     .core_chain_locked_height()
                     - transaction_sign_height;
 
-                let status;
+                let is_chain_locked =
+                    *core_transactions_statuses
+                        .get(&transaction_id)
+                        .ok_or(Error::Execution(ExecutionError::CorruptedCodeExecution(
+                            "we should always have a withdrawal status",
+                        )))?;
+
+                let mut status = current_status;
 
-                if core_transactions.contains(&transaction_id) {
-                    status = withdrawals_contract::WithdrawalStatus::COMPLETE;
+                if is_chain_locked {
+                    status = WithdrawalStatus::COMPLETE;
                 } else if block_height_difference > NUMBER_OF_BLOCKS_BEFORE_EXPIRED {
-                    status = withdrawals_contract::WithdrawalStatus::EXPIRED;
-                } else {
-                    return Ok(None);
```

Comment:
Without this condition it will update all documents every block even if nothing changed.

Reply:
We have a cycling issue if we return, and we have an update too much issue if we don't

Reply:
(from call with Ivan)